### PR TITLE
Feature/documentation updates

### PR DESCRIPTION
**Pull request by dgergel**

### CODE REVIEW

File: docs/Development/ReleaseNotes.md [hunk 1]
```diff
@@ -27,44 +27,54 @@ This is a major update from VIC 4. The VIC 5.0.0 release aims to have nearly ide
...
 
 	A set of logging Macros have been added to all drivers and `vic_run`. The logging level can be set in the driver `Makefile` via the `LOG_LVL` variable. The logging Macros provide the filename and line number in the source code to aid in debugging.  Additionally, when compiler support is available, a traceback is printed when VIC exits during runtime. When the `LOG_DIR` variable is provided in the global parameter file, VIC will write its log(s) to log files instead of printing to stdout.
 
+
 6. Sub-hourly Timestep ([GH#188](https://github.com/UW-Hydro/VIC/pull/188))
 
-	Previous versions of VIC were limited to a minimum timestep of one hour. The units of the VIC timestep have been changed from hours to seconds and the minimum timestep is now one second. If you intend on running VIC at a timestep less that one hour, we suggest significant testing.
+	Previous versions of VIC were limited to a minimum timestep of one hour. The units of the VIC timestep have been changed from hours to seconds and the minimum timestep is now one second. If you intend on running VIC at a timestep of less than one hour, we suggest significant testing.
+
```
Comment (bartnijssen) [inline docs/Development/ReleaseNotes.md]: s/significant/extensive/

File: docs/Development/Testing.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ The VIC Test Suite includes six main test types:
     * restart:  tests that address model state and restart capability.
     * I/O:  tests that address model input and output functionality.
         *  forcings come out the way they come in
-        *  parameter files are appropriately read in allocated.
+        *  parameter files are appropriately read in and allocated.
```
Comment (bartnijssen) [inline docs/Development/Testing.md]: s/read in/ingested/

File: docs/Development/working-with-git.md [hunk 3]
```diff
@@ -20,15 +20,15 @@ In general, if you are only going to be using the model and not working directly
 If you plan on contributing to model development or would like a systematic way to incorporate updates to the VIC source code, we encourage you to use Git.  The following sections are designed to get you started using git and working with the VIC source code repository.
 
 ## Git Resources
-If you are not familiar with [Git](http://git-scm.com/) yet, we encourage you to spend a few minutes getting antiquated with the system before you starting working with the VIC source code and Git.  It's not difficult to use and a few minutes of learning about [Git](http://git-scm.com/) will go along way in helping you manage your code development.
```
Comment (bartnijssen) [inline docs/Development/working-with-git.md]: That's hilarious. I know that some people certainly feel that any time with git makes them age prematurely

File: docs/Development/working-with-git.md [hunk 4]
```diff
@@ -20,15 +20,15 @@ In general, if you are only going to be using the model and not working directly
...
+If you are not familiar with [Git](http://git-scm.com/) yet, we encourage you to spend a some time becoming acquainted with the system before you starting working with the VIC source code and Git.  A few minutes of learning about [Git](http://git-scm.com/) will go along way in helping you manage your code development.
```
Comment (bartnijssen) [inline docs/Development/working-with-git.md]: "we encourage you to spend a some time becoming acquainted with the system before you starting working with the VIC source code and Git."

suggested change

"we encourage you to get acquainted with it before you starting working with the VIC source code."

File: docs/Development/working-with-git.md [hunk 5]
```diff
@@ -20,15 +20,15 @@ In general, if you are only going to be using the model and not working directly
...
 
 There are a number of good [Git](http://git-scm.com/) learning resources that will provide a basic introduction to the version control system.
 
 - http://git-scm.com/about
 - https://help.github.com/
 
 # Getting the code
-The basics steps to get the VIC source code repository are as follows.  This is basically a VIC specific rendition of https://help.github.com/articles/fork-a-repo
+The basic steps to obtain the VIC source code repository are as follows.  This follows the same steps as the following tutorial: https://help.github.com/articles/fork-a-repo
```
Comment (bartnijssen) [inline docs/Development/working-with-git.md]: "The basic steps to obtain the VIC source code repository are as follows.  This follows the same steps as the following tutorial:"

suggested change

"To obtain the VIC source code repository follow the step in this tutorial:"

**Comment by dgergel** [comment]
Updated to reflect @bartnijssen's suggested changes.

**Pull request merged**